packages/instrumentation-tracing-web/src/index.ts

```diff
@@ -0,0 +1,132 @@
+import type { ReadableSpan, SpanExporter, SpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { WebTracerProvider } from '@opentelemetry/sdk-trace-web';
+
+class GrafanaSpanProcessor implements SpanProcessor {
+  private activeSpan: Span | null = null;
+  private isShutdown = false;
+  private shuttingDownPromise = Promise.resolve();
+
+  constructor(private exporter: SpanExporter) {}
+
+  forceFlush(): Promise<void> {
+    // spans are returned without waiting
+    return Promise.resolve(undefined);
+  }
+
+  onStart(span: Span, _parentContext: Context): void {
+    this.activeSpan = span;
+  }
+
+  onEnd(span: ReadableSpan): void {
+    if (this.isShutdown) {
+      return;
+    }
+
+    if (this.activeSpan?.spanContext().spanId === span.spanContext().spanId) {
+      this.activeSpan = null;
+    }
+
+    if ((span.spanContext().traceFlags & TraceFlags.SAMPLED) === 0) {
+      return;
+    }
+
+    context.with(suppressTracing(context.active()), () => {
+      this.exporter.export([span], (result) => {
+        const { code, error } = result;
+
+        if (code !== ExportResultCode.SUCCESS) {
+          globalErrorHandler(error ?? new Error(`SimpleSpanProcessor: span export failed (status ${result})`));
+        }
+      });
+    });
+  }
+
+  shutdown(): Promise<void> {
+    if (this.isShutdown) {
+      return this.shuttingDownPromise;
+    }
+
+    this.isShutdown = true;
+
+    this.shuttingDownPromise = new Promise((resolve, reject) => {
+      Promise.resolve()
+        .then(() => this.exporter.shutdown())
+        .then(resolve)
+        .catch(reject);
+    });
+
+    return this.shuttingDownPromise;
+  }
+
+  getActiveSpan(): Span | null {
+    return this.activeSpan;
+  }
+}
+
+class GrafanaSpanExporter implements SpanExporter {
+  export(spans: ReadableSpan[], resultCallback: (result: ExportResult) => void): void {
+    spans
+      .map((span) => ({
+        traceId: span.spanContext().traceId,
+        parentId: span.parentSpanId,
+        name: span.name,
+        id: span.spanContext().spanId,
+        kind: span.kind,
+        timestamp: hrTimeToMicroseconds(span.startTime),
+        duration: hrTimeToMicroseconds(span.duration),
+        attributes: span.attributes,
+        status: span.status,
+        events: span.events,
+      }))
+      .forEach((span) => {
+        agent.api.callOriginalConsoleMethod(LogLevel.LOG, span);
+      });
+
+    resultCallback({ code: ExportResultCode.SUCCESS });
+  }
+
+  shutdown(): Promise<void> {
+    return Promise.resolve(undefined);
+  }
+}
+
+const tracingInstrumentation: Instrumentation = () => {
+  const provider = new WebTracerProvider();
+  provider.addSpanProcessor(new GrafanaSpanProcessor(new GrafanaSpanExporter()));
+  provider.register({
+    propagator: new W3CTraceContextPropagator(),
+    contextManager: new ZoneContextManager(),
+  });
+
+  registerInstrumentations({
+    instrumentations: [
```

Comment:
I think we should allow user to overide instrumentations, propagator and add processors via options